book/src/dev/rfcs/xxxx-block-subsidy.md

```diff
@@ -0,0 +1,304 @@
+
+# Motivation
+[motivation]: #motivation
+
+All incoming blocks must be validated with the protocol specifications, there is no way to avoid including this logic in any Zcash protocol compatible implementation such as Zebra.
+
+Block subsidy and miner fees are part of the protocol, the semantic verification of them apply to the coinbase transaction and they must be verified in Zebra.
+
+# Definitions
+[definitions]: #definitions
+
+- **block subsidy**: Value created from mining composed of miner subsidy plus founders reward or funding stream if any of the 2 are active ([Subsidy Concepts](https://zips.z.cash/protocol/protocol.pdf#subsidyconcepts)).
+- **coinbase transaction**: The first transaction in a block; this is the transaction that handles block subsidy.
+- **founders reward**: The portion of the block subsidy that goes into a pre defined founder address in a single output.
+- **funding streams**: The portion of the block subsidy that goes into one or more pre defined funding stream addresses. Payment is done with one output for each active funding stream of the block.
+- **miner subsidy**: The portion of the block subsidy that goes into the miner of the block, excluding fees.
+- **network upgrade**: An intentional consensus rule change undertaken by the community in order to improve the network.
+- **transaction fees**: The sum of the extra [transparent value pool](#transparent-value-pool-calculation) and shielded values, for all the transactions in a block. This amount can be spent by the miner in the coinbase transaction.
+
+### Transaction fees calculation
+
+There is a value pool inside each block called "[chain value pool balance](https://zips.z.cash/zip-0209#terminology)". Balance of that pool works as follows:
+
+Transparent:
+
+- `tx_in` adds to the pool.
+- `tx_out` subtracts from the pool.
+
+https://zips.z.cash/protocol/protocol.pdf#txnencodingandconsensus
+
+Sprout:
+
+- `vpub_new` values from `vJoinSplit` of the transaction add to the pool.
+- `vpub_old` values from `vJoinSplit` of the transaction subtract from the pool.
+
+https://zips.z.cash/protocol/protocol.pdf#joinsplitencodingandconsensus
+
+Sapling:
+
+- `valueBalance` adds to the pool. It itself is equal to `sum(SaplingSpends) - sum(SaplingOutputs)`, so it has the same functional effect as for transparent and Sprout.
+
+https://zips.z.cash/protocol/protocol.pdf#txnencodingandconsensus
+
+The balance rule is that this pool must have non-negative value(See [ZIP 209](https://zips.z.cash/zip-0209)), and its net value is the miner fee for the transaction.
```

Comment:
Let's do the extra checks recommended by Daira:
```suggestion
The balance rule is that this pool must have non-negative value (see [ZIP 209](https://zips.z.cash/zip-0209)), and its net value is the miner fee for the transaction. Non-negative transparent and block value pool balances are implied by the other consensus rules. Zebra should check or assert that these balances are non-negative, so invalid blocks don't pass verification. (And so that we guard against implementation bugs.)
```